<launch>

  <arg name="min_count" default="13" />
  <arg name="disparity_threshold" default="2" />

  <node pkg="disparity_to_point_cloud" type="disparity_to_point_cloud_node" name="disparity_to_point_cloud" output="screen">
    <remap to="/cam1_image" from="/disparity"/>
    <remap to="/omi_cam/point_cloud" from="/point_cloud"/>
    <param name="min_count" value="$(arg min_count)" />
    <param name="disparity_threshold" value="$(arg disparity_threshold)" />
  </node>

  <!-- <node name="tf_fix_broad" pkg="disparity_to_point_cloud" type="fix_tf.py" respawn="false" output="screen" ></node> -->

  <node name="message_republisher" pkg="ait_ros_messages" type="omni_vio_sensor_republisher.py" respawn="false" output="screen" ></node>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
    <param name="resolution" value="0.1" />
          <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
          <param name="frame_id" type="string" value="world" />
          <!-- maximum range to integrate (speedup!) -->
          <param name="sensor_model/max_range" value="50.0" />
          <!-- data source to integrate (PointCloud2) -->
          <remap from="cloud_in" to="/omi_cam/point_cloud" />
   </node>

<node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find disparity_to_point_cloud)/rviz/d2pc.rviz" />

</launch>
